native/src/main/java/io/ballerina/lib/data/csvdata/csv/CsvParser.java

```diff
@@ -316,10 +351,31 @@ public Object execute(Reader reader, Type type, CsvConfig config, BTypedesc bTyp
             try {
                 char[] buff = new char[1024];
                 int count;
+                AtomicBoolean isOverwritten = new AtomicBoolean(false);
                 while ((count = reader.read(buff)) > 0) {
```

Comment:
Don't we need to manually close this reader when there is an exception and we exit immediately?